transportable-udfs-compile-utils/src/main/java/com/linkedin/transport/compile/TransportUDFMetadata.java

```diff
@@ -64,50 +74,49 @@ public static TransportUDFMetadata fromJsonFile(File jsonFile) {
   }
 
   public static TransportUDFMetadata fromJson(Reader reader) {
-    return TransportUDFMetadataSerDe.toUDFMetadata(GSON.fromJson(reader, TransportUDFMetadataJson.class));
+    return TransportUDFMetadataSerDe.deserialize(new JsonParser().parse(reader));
   }
 
-  /**
-   * Represents the JSON object structure of the Transport UDF metadata resource file
-   */
-  private static class TransportUDFMetadataJson {
-    private List<UDFInfo> udfs;
+  private static class TransportUDFMetadataSerDe {
 
-    TransportUDFMetadataJson() {
-      this.udfs = new LinkedList<>();
+    public static TransportUDFMetadata deserialize(JsonElement json) {
```

Comment:
Any reason to manually serialize/deserialize instead of letting GSON do it as per the schema?

Reply:
I tried it and it would be more complex, especially with the need to track class parameters.